### PR TITLE
Shop search

### LINKED ISSUE #34: Feature: Shop searching
We want to be able to search for a shop based on its title and its categories

### PR DESCRIPTION
**Pull request by simonkrol**
Adds the functionality for searching for shops either by category or by tag. If only a single shop exists, search will go directly to its show page, if more than one exist, it'll direct you to the index with only those shops listed.

Resolves #34

### CODE REVIEW

**Review by BoyanSiromahov — APPROVED** [review]
+1

File: src/main/java/model/CategoryRepository.java [hunk 1]
```diff
@@ -2,8 +2,10 @@
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.data.rest.core.annotation.RepositoryRestResource;
 
+import java.util.List;
```
Comment (L-Murphy) [inline src/main/java/model/CategoryRepository.java:5]: Nice job importing!! A+

File: src/test/java/model/SearchIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
```
Comment (L-Murphy) [inline src/test/java/model/SearchIntegrationTest.java:13]: Woah two imports! Now that's big-league stuff!

**Review by L-Murphy — APPROVED** [review]
Nice

**Pull request merged**